gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -17,41 +17,24 @@
 // <http://www.gnu.org/licenses/>.
 
 #include "rust-macro-builtins.h"
+#include "rust-ast-fragment.h"
 #include "rust-ast.h"
+#include "rust-attribute-visitor.h"
 #include "rust-diagnostics.h"
+#include "rust-early-name-resolver.h"
 #include "rust-expr.h"
-#include "rust-session-manager.h"
-#include "rust-macro-invoc-lexer.h"
 #include "rust-lex.h"
+#include "rust-macro-invoc-lexer.h"
+#include "rust-macro.h"
 #include "rust-parse.h"
-#include "rust-early-name-resolver.h"
-#include "rust-attribute-visitor.h"
+#include "rust-session-manager.h"
+#include <memory>
+#include <utility>
```

Comment:
GCC does not like when you include these system headers and you should not have to. You can include "rust-system.h" if you get include failures